CRM/Custom/Form/Field.php

```diff
@@ -315,26 +315,22 @@ public function buildQuickForm() {
       $this->add('checkbox', 'in_selector', ts('Display in Table?'));
     }
 
-    if ($this->_action == CRM_Core_Action::UPDATE) {
-      $this->freeze('data_type');
-    }
+    $optionGroupParams = [
+      'is_reserved' => 0,
+      'is_active' => 1,
+      'options' => ['limit' => 0, 'sort' => "title ASC"],
+      'return' => ['title'],
+    ];
 
     if ($this->_action == CRM_Core_Action::UPDATE) {
-      $optionGroupParams = [
-        'id' => $this->_values['option_group_id'],
-        'return' => ['title'],
-      ];
-    }
-    else {
-      $optionGroupParams = [
-        'is_reserved' => 0,
-        'is_active' => 1,
-        'options' => ['limit' => 0, 'sort' => "title ASC"],
-        'return' => ['title'],
-      ];
+      $this->freeze('data_type');
+      // Before dev/core#155 we didn't set the is_reserved flag properly, which should be handled by the upgrade script...
+      //  but it is still possible that existing installs may have optiongroups linked to custom fields that are marked reserved.
+      $optionGroupParams['id'] = $this->_values['option_group_id'];
+      $optionGroupParams['options'] = ['or' => [["is_reserved", "id"]]];
```

Comment:
Need to use the `+=` operator here to avoid clobbering existing options.

Reply:
Oops. updated :-)